Tests/Admin/AdminTest.php

```diff
@@ -1456,21 +1465,22 @@ public function testRemoveFieldFromFormGroup()
                 ),
             ),
         );
-
-        $admin = new PostAdmin('sonata.post.admin.post', 'Application\Sonata\NewsBundle\Entity\Post', 'SonataNewsBundle:PostAdmin');
         $admin->setFormGroups($formGroups);
+        $this->assertEquals($admin->getFormGroups(), $formGroups);
 
-        $admin->removeFieldFromFormGroup('foo');
```

Comment:
Deprecated method should keep being tested.

I propose:

* keep the `testRemoveFieldFromFormGroup` as is and add a `@group legacy` on doc comment.
* create a new test named `testDefineFormBuilderCleanupFormGroupsAndTabs` with your logic inside.